homeassistant/components/roborock/__init__.py

```diff
@@ -0,0 +1,65 @@
+"""The Roborock component."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from roborock.api import RoborockApiClient
+from roborock.cloud_api import RoborockMqttClient
+from roborock.containers import HomeDataDevice, RoborockDeviceInfo, UserData
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import CONF_BASE_URL, CONF_USER_DATA, DOMAIN, PLATFORMS
+from .coordinator import RoborockDataUpdateCoordinator
+
+SCAN_INTERVAL = timedelta(seconds=30)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up roborock from a config entry."""
+    _LOGGER.debug("Integration async setup entry: %s", entry.as_dict())
+
+    user_data = UserData.from_dict(entry.data[CONF_USER_DATA])
+    api_client = RoborockApiClient(entry.data[CONF_USERNAME], entry.data[CONF_BASE_URL])
+    _LOGGER.debug("Getting home data")
+    home_data = await api_client.get_home_data(user_data)
+    _LOGGER.debug("Got home data %s", home_data)
+    devices: list[HomeDataDevice] = home_data.devices + home_data.received_devices
+    # Create a mqtt_client, which is needed to get the networking information of the device for local connection and in the future, get the map.
+    mqtt_client = RoborockMqttClient(
+        user_data, {device.duid: RoborockDeviceInfo(device) for device in devices}
+    )
+    network_info = {
+        device.duid: await mqtt_client.get_networking(device.duid) for device in devices
```

Comment:
Sorry for the late comment -- both here and in the update coordinator, when doing multiple serial awaits the preference is instead to use `asyncio.gather`

Reply:
No need to apologize. I've made the change